### PR TITLE
DimensionsForm: Use ipl-form

### PR DESCRIPTION
**Pull request by sukhwinder33445**
require https://github.com/Icinga/ipl-html/pull/116

This PR also fixes the problem that the search bar filters are deleted when adding, moving or removing dimensions.

### CODE REVIEW

File: library/Cube/Web/IdoController.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 namespace Icinga\Module\Cube\Web;
 
+use GuzzleHttp\Psr7\ServerRequest;
```
Comment (nilmerg) [inline library/Cube/Web/IdoController.php:7]: Is it possible to use the CompatController as base here? I don't particularly like to have this explicit dependency here.

File: application/forms/DimensionsForm.php [hunk 2]
```diff
@@ -7,11 +7,22 @@
...
 use Icinga\Web\Notification;
+use ipl\Html\Form;
+use ipl\Html\Html;
+use ipl\Web\Common\FormUid;
+use ipl\Web\Url;
+use ipl\Web\Widget\Icon;
 
 class DimensionsForm extends Form
 {
+    use FormUid;
```
Comment (nilmerg) [inline application/forms/DimensionsForm.php:19]: Also use the `Translation` trait. There will be change soon in ipl-html which causes it to not being part of `ipl\Html\Form` anymore.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: application/forms/DimensionsForm.php [hunk 3]
```diff
@@ -7,11 +7,25 @@
...
+    use Translation;
+
+
```
Comment (nilmerg) [inline application/forms/DimensionsForm.php:23]: Single empty lines please.

File: library/Cube/Web/IdoController.php [hunk 4]
```diff
@@ -97,10 +103,6 @@ private function prepareCube(): void
             }
         }
 
-        foreach (['renderLayout', 'showFullscreen', 'showCompact', 'view'] as $p) {
-            $this->params->shift($p);
-        }
-
```
Comment (nilmerg) [inline library/Cube/Web/IdoController.php:103]: Why this change?

**Review by nilmerg — CHANGES_REQUESTED** [review]
Please also highlight remove buttons once focused. Just like when hovered, but plus an outline.

File: library/Cube/Web/IdoController.php (hunk 4 above)
Comment (sukhwinder33445) [reply library/Cube/Web/IdoController.php:103]: These are superfluous parameters, which now the CompatController already removes.

Comment (nilmerg) [reply library/Cube/Web/IdoController.php:103]: :bulb:

File: public/css/module.less [hunk 5]
```diff
@@ -245,6 +245,15 @@ fieldset.dimensions {
...
       color: @text-color-on-icinga-blue;
     }
+
+    &:focus {
+      outline: 3px solid fade(@icinga-blue, 50%);
+      outline-offset: 1px;
+
+      .rounded-corners();
+      background-color: @icinga-blue;
+      color: @text-color-on-icinga-blue;
```
Comment (nilmerg) [inline public/css/module.less:255]: Why did you repeat this? Just extend the rule above, please.

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Comment by nilmerg** [comment]
The change regarding `hasBeenSubmitted` in ipl-html is now postponed. Override and patch it yourself in `DimensionsForm`, I'd say.

**Review by nilmerg — APPROVED** [review]
